src/matter/interaction/InteractionMessenger.ts

```diff
@@ -194,7 +194,7 @@ export class InteractionServerMessenger extends InteractionMessenger<MatterDevic
         if (dataReport.suppressResponse) {
             // We do not expect a response other than a Standalone Ack, so if we receive anything else, we throw an error
             await tryCatchAsync(async () => await this.exchange.send(MessageType.ReportData, TlvDataReport.encode(dataReport), true), UnexpectedMessageError, error => {
-                const { payloadHeader: { messageType }, payload } = (error as UnexpectedMessageError).receivedMessage;
+                const { payloadHeader: { messageType }, payload } = error.receivedMessage;
```

Comment:
nit, you can also de-structure receivedMessage
const { receivedMessage: { payloadHeader: { messageType }, payload } }= error;

or even better, you can de-structure in the method argument:
( { receivedMessage: { payloadHeader: { messageType }, payload } }) => this.throwIfError(messageType, payload)

or even better, you can modify throwIfError to take the message as an argument:
({ receivedMessage }) => this.throwIfError(receivedMessage)
(and probably throwIfError should be renamed to throwIfErrorStatusMessage)

Reply:
Changed to last proposal and also adjusted in second throwIfError case (SecureChannelmessenger) accordingly